WcaOnRails/app/models/competition_event.rb

```diff
@@ -59,6 +67,8 @@ def load_wcif!(wcif)
     self.rounds = new_rounds
     WcifExtension.update_wcif_extensions!(self, wcif["extensions"]) if wcif["extensions"]
     self
+    self.qualification = Qualification.load(wcif["qualification"])
+    self.save
```

Comment:
Without this, hitting "save" does not actually save qualification times -- they're updated on `self` but nothing makes `self` get written to DB. This reproduces with spec/models/competition_wcif_spec.rb:461.

I may not be remembering this fully, but my rough understanding is that all of the other attributes here are standalone DB objects, while qualification is the only one that's saved directly in Events. ActiveRecord is able to figure out that `self.rounds = new_rounds` means that the rounds in `new_rounds` need to be committed, but not that setting `self.qualification` means that `self` needs to be committed.

I don't fundamentally understand why ActiveRecord doesn't write the event, though.

Reply:
At first glance, I don't think this is something ActiveRecord is ever able to "figure out". Instead, you can see in L63 that the round is written to the database "manually", and so are the extensions. Could you do a similar thing for the qualification, instead of taking the current "brute force save" approach?

Reply:
Ah -- like this?

Reply:
Yes, although now I'm debating with myself which of the versions is cleaner.
No strong preference here, but perhaps the old `save` approach is a bit more concise despite its generic "sledgehammer" approach...=
I'll leave it up to you to decide whether you want to keep the current `update` or whether to revert to `save`